internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -87,46 +87,45 @@ spec:
           TMPFILE=$(mktemp)
           PYXIS_URL="https://pyxis.engineering.redhat.com/v1"
 
-          IFS="/" read REGISTRY REPO IMAGE <<< "${1}"
-          IFS=":" read IMAGE TAG <<< "${IMAGE}"
+          IFS="/" read -r REGISTRY REPO IMAGE <<< "${1}"
+          IFS=":" read -r IMAGE TAG <<< "${IMAGE}"
 
           FETCH_URL="${PYXIS_URL}/repositories/registry/${REGISTRY}/repository/${REPO}/${IMAGE}/tag/${TAG}"
 
           # strips the last "/tag" in case $TAG is not set
           [ -z "${TAG}" ] && FETCH_URL=${FETCH_URL%/tag*}
 
-          curl --negotiate -u: "${FETCH_URL}" -o $TMPFILE
+          curl --negotiate -u: "${FETCH_URL}" -o "${TMPFILE}"
           # prints "false" in case .fbc_opt_in entry is missing
-          jq -e -r '.fbc_opt_in //false' $TMPFILE && rm -f $TMPFILE
+          jq -e -r '.fbc_opt_in //false' "${TMPFILE}" && rm -f "${TMPFILE}"
+        }
+
+        # checks if there is any previous build for the same snapshot.
+        # in case multiple builds are found, returns only the last one.
```

Comment:
When found, it is always a list with at least 1 element. Normally, just one, but as I was rebuilding over and over the same fbc_fragment, it ended up with more than one, so I added the index to make sure it should always get something . My initial implementation I was fetching them from `items[]`.

If the code ignore the failed, the tendency is that it will never return more than one, because we would return only in progress or complete, so the fbc_fragment will have to change to allow a new build.

You can see it here: [here](https://iib.stage.engineering.redhat.com/api/v1/builds?user=rhtap-release-iib-stage@IPA.REDHAT.COM&from_index=registry-proxy.engineering.redhat.com/rh-osbs/iib-preview-rhtap:v4.12)

Reply:
But maybe you are right. I might be considering the wrong one depending on what the request got. I will crosscheck this.

Reply:
Another comment. The link above uses to search the "from_index"  used by e2e, so the fbc_fragment is always the same. Not sure how frequent this would happen for actual users.

Reply:
Thanks for the link to iib. Ok, so at least I can see that the order is latest to oldest. So yeah, we want something from the top. But I still don't see how this filter could work properly:

```
jq -r --arg fbc_fragment "${fbc_fragment}" '.items[0] |select(.fbc_fragment==$fbc_fragment)'
```

You only select the 1 latest build from the results. And then you do the select - so if it's the one with your fragment, the build json is the result, if not, the result is empty. And it doesn't even do anything to skip failed builds - how is that supposed to work?

You will need something like:
```
jq '[.items[] |select(.fbc_fragment=="quay.io/hacbs-release-tests/fbc-sample-repo@sha256:857814679c1deec5bc5d6d8064832b4c0af40dcb07dad57c48f23e5ba6926aed" and .state != "failed")][0]'
```

(Not sure what the state value for failed is.)

And then you need to check if that result is empty or not - if not, you can use that build, otherwise you need a new one.